libraries/core_libs/node/src/node.cpp

```diff
@@ -226,6 +206,39 @@ void FullNode::start() {
         },
         *rpc_thread_pool_);
   }
+  subscription_pool_ = std::make_unique<util::ThreadPool>(1);
+
+  // GasPricer updater
+  final_chain_->block_finalized_.subscribe(
+      [gas_pricer = as_weak(gas_pricer_)](auto const &res) {
+        if (auto _gas_pricer = gas_pricer.lock()) {
+          _gas_pricer->update(res->trxs);
+        }
+      },
+      *subscription_pool_);
```

Comment:
why does block_finalized_ use subscription_pool_ and block_applying_  does not ?

Reply:
because block_applying_ is wrongly written

Reply:
is it too difficult to adjust it so all event use this subscription_pool_ ?

Reply:
probably not, but I do not want to touch HF as it is sensitive, I would rather create separate task for it. I think it needs a lot of testing that is out of the scope of this task

Reply:
https://github.com/Taraxa-project/taraxa-node/issues/1644